File: Web/AppServer/AppServer/Models/Device.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace AppServer.Models
{
    /// <summary>
    /// Device entity, representing device able to send or receive signals
    /// </summary>
    public class Device : IEntityBase
    {
        private List<Signal> _signals = new List<Signal>();

        [Required]
        public int ID { get; set; }
        [Required]
        public string Name { get; set; }
        [Required]
        public DeviceCommunicationMethod CommunicationMethod { get; set; }

        public List<Signal> Signals { get { return _signals; } }

        public override bool Equals(object obj)
        {
            if (obj == null)
                throw new ArgumentNullException();

            Device device = obj as Device;
            if (device == null)
                return base.Equals(obj);

            return device.ID == ID;
        }

        public override int GetHashCode()
        {
            return base.GetHashCode();
        }
    }
}

File: Web/appserver/src/main/java/com/appserver/models/Device.java
package com.appserver.models;

import java.util.ArrayList;
import java.util.List;

/**
 * Device entity, representing device able to send or receive signals
 */
public class Device extends EntityBase
{
	private int ID = 0;
	private String Name;
	private DeviceCommunicationMethod CommunicationMethod;
	private String Icon;
	private List<Signal> Signals = new ArrayList<Signal>();
	private int SignalSourceID = 0;
	private int SignalDestinationID = 0;

    public int getID() { return ID; }
	public void setID(int id) { ID = id; }

    public String getName() { return Name; }
	public void setName(String name) { Name = name; }
	
	public String getIcon() { return Icon; }
	
	public DeviceCommunicationMethod getCommunicationMethod() {	return CommunicationMethod;	}
	public void setCommunicationMethod(DeviceCommunicationMethod communicationMethod) {	CommunicationMethod = communicationMethod; }
	public List<Signal> getSignals() {	return Signals;	}
	
    public int getSignalSourceID()
    {
    	if (SignalSourceID == 0)
    		return ID;
    	return SignalSourceID;
    }
	public void setSignalSourceID(int id) { SignalSourceID = id; }	
	
    public int getSignalDestinationID()
    {
    	if (SignalDestinationID == 0)
    		return ID;
    	return SignalDestinationID;
    }
	public void setSignalDestinationID(int id) { SignalDestinationID = id; }	
	
	public Device(int id)
	{
		ID = id;
	}	
}
